src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component;
+
+/**
+ * Interface Factory
+ *
+ * @package ILIAS\UI\Component\Modal
+ */
+interface Factory
+{
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *      An Interruptive Modal disrupts the user in critical situation, forcing him or her to focus on the task at hand.
+     *   composition: >
+     *      The Modal states why this situation needs attention and may point out consequences.
+     *   effect: >
+     *      All controls of the original context are inaccessible until the Modal is completed. Upon completion the user returns to the original context.
+     *
+     * rules:
+     *   usage:
+     *     1: >
+     *        Due to the heavily disruptive nature of this type of Modal it MUST be restricted to critical situations (e.g. loss of data).
+     *     2: >
+     *        All delete-actions are critical situations and MUST be implemented as an Interruptive Modal.
+     *     3: >
+     *        Interruptive Modals MUST contain a Default Button continuing the action that initiated the Modal (e.g. Delete the Item) on the left side of the footer of the modal and a button canceling the action on the right side of the footer as Default Button.
+     *     4: >
+     *        The Cancel Button in the footer and the Close Button in the header MUST NOT perform any additional action than closing the confirmation screen.
+     * ---
+     * @param string $title
+     * @param Component\Component $content
+     * @return \ILIAS\UI\Component\Modal\Interruptive
+     */
+    public function interruptive($title, Component\Component $content);
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     Round-Trip Modals are to be used if the context would be lost by performing this action otherwise. Round-Trip Modals accommodate sub-workflows within an overriding workflow. The Round-Trip Modal ensures that the users does not leave the trajectory of the overriding workflow. This is typically the case if an ILIAS service is called for while in working in an object.
+     *   composition: >
+     *     Round-Trip Modals are completed by a small well defined sequence of 2 steps max. that might be displayed on a sequence of different modals connected through some “next” Button.
```

Comment:
But the number would also depend on what the user does during the workflow. The modal would not be aware, since we just would quietly replace its content 1-n times.